tests/rptest/clients/rpk.py

```diff
@@ -1303,3 +1313,58 @@ def cloud_byoc_aws_apply(self, redpanda_id, token, extra_flags=[]):
 
         out = self._execute(cmd, env=envs)
         return json.loads(out)
+
+    def cluster_health(self):
+        cmd = [
+            self._rpk_binary(), "-X", "admin.hosts=" + self._admin_host(),
+            "cluster", "health"
+        ]
+        out = self._execute(cmd)
+
+        # CLUSTER HEALTH OVERVIEW
+        # =======================
+        # Healthy:                     true
+        # Unhealthy reasons:           []
+        # Controller ID:               0
+        # All nodes:                   [0 1 2 3 4]
+        # Nodes down:                  []
+        # Leaderless partitions:       []
+        # Under-replicated partitions: []
+        def parse_field(field_name, string):
+            pattern = re.compile(f"^{field_name}: +(?P<value>.+)")
+            m = pattern.match(string)
+            assert m is not None, f"Field string '{string}' does not match the pattern"
+            return m['value']
+
+        def parse_list(list_str):
+            if list_str == "[]":
+                return []
+
+            elements = list_str.strip('[]').split()
+            parsed_list = []
+
+            if all(e.isdigit() for e in elements):
+                parsed_list = [int(e) for e in elements]
+            else:
+                parsed_list = [f"'{e}'" for e in elements]
```

Comment:
If the values in `elements` do not satisfy `isdigit()`, then those values are already strings. Therefore, I believe you do not need the single quotes (') or double quotes (") here.
```suggestion
            parsed_list = elements

            if all(e.isdigit() for e in elements):
                parsed_list = [int(e) for e in elements]
```

Reply:
Thanks! resolved in the last force push: https://github.com/redpanda-data/redpanda/compare/57c1419158f7ccca886d667846b4fa2d472e0f77..0e482386aa96bd16e89198c86d9f032d9698c182